pkg/debug/debug.go

```diff
@@ -83,6 +84,7 @@ func initDebugLogger() {
 	}
 
 	opts.logFile = f
+	LogWriter = f
```

Comment:
Initializing this only in the debug log enabled case would lead to segfaults.

But after #2581 is merged you can rebase and we can merge your changes to encrypt.go.